### ISSUE TITLE
ImportError: cannot import name 'inference_segmentor' from 'mmseg.apis' 

### ISSUE THREAD

**Issue by nnmaitian** [TencentARC/DeSRA#9]
Hello, I encountered the following problem, from mmseg.apis import inference_segmentor, init_segmentor, show_seg
ImportError: cannot import name 'inference_segmentor' from 'mmseg.apis' . How to solve it.

**Comment by 201210302** [comment]
change the function name in artifact_detection.py
![스크린샷 2023-12-08 123140](https://github.com/TencentARC/DeSRA/assets/90252213/5895c6ea-e0ec-45bd-ad0d-8b8574e92cf9)

#from mmseg.apis import inference_segmentor, init_segmentor, show_seg
from mmseg.apis import inference_model, init_model, show_result_pyplot
#from mmseg.core.evaluation import get_palette
from mmseg.utils import get_palette

**Comment by LiangbinXie** [comment]
@nnmaitian Did you follow the specific version of mmsegmentation?

**Comment by nnmaitian** [comment]
> change the function name in artifact_detection.py ![스크린샷 2023-12-08 123140](https://private-user-images.githubusercontent.com/90252213/288954526-5895c6ea-e0ec-45bd-ad0d-8b8574e92cf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NjA2MTQsIm5iZiI6MTcwMjQ2MDMxNCwicGF0aCI6Ii85MDI1MjIxMy8yODg5NTQ1MjYtNTg5NWM2ZWEtZTBlYy00NWJkLWFkMGQtOGI4NTc0ZTkyY2Y5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDA5MzgzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2MDZhZjJiYmRmZjlkYTI4MzliYmRjNmE2MTkxMGVlMTVlZWZlZjI5ZDAxN2I2OWQ0ZmZjN2I3ZmQxNzljNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.D8NaDBX-QKBhv1zzuiJIsdUqRLmX6xIxoHBZu8Ny_0M)
> 
> #from mmseg.apis import inference_segmentor, init_segmentor, show_seg from mmseg.apis import inference_model, init_model, show_result_pyplot #from mmseg.core.evaluation import get_palette from mmseg.utils import get_palette



> the specific version of mmsegmentation?

Yes. I use the following code to specify the mmseg version and installation.

git reset --hard 356215bf586b1404d4cb0c205685fcf589df6148
pip install -v -e .
pip install mmcv-full==1.6.1

But still ImportError: cannot import name 'show_seg' from 'mmseg.apis'

**Comment by nnmaitian** [comment]
> @nnmaitian Did you follow the specific version of mmsegmentation?

Yes. I use the following code to specify the mmseg version and installation.

git reset --hard 356215bf586b1404d4cb0c205685fcf589df6148
pip install -v -e .
pip install mmcv-full==1.6.1

But still ImportError: cannot import name 'show_seg' from 'mmseg.apis'

**Comment by nnmaitian** [comment]
> change the function name in artifact_detection.py ![스크린샷 2023-12-08 123140](https://private-user-images.githubusercontent.com/90252213/288954526-5895c6ea-e0ec-45bd-ad0d-8b8574e92cf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NjA2MTQsIm5iZiI6MTcwMjQ2MDMxNCwicGF0aCI6Ii85MDI1MjIxMy8yODg5NTQ1MjYtNTg5NWM2ZWEtZTBlYy00NWJkLWFkMGQtOGI4NTc0ZTkyY2Y5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDA5MzgzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2MDZhZjJiYmRmZjlkYTI4MzliYmRjNmE2MTkxMGVlMTVlZWZlZjI5ZDAxN2I2OWQ0ZmZjN2I3ZmQxNzljNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.D8NaDBX-QKBhv1zzuiJIsdUqRLmX6xIxoHBZu8Ny_0M)
> 
> #from mmseg.apis import inference_segmentor, init_segmentor, show_seg from mmseg.apis import inference_model, init_model, show_result_pyplot #from mmseg.core.evaluation import get_palette from mmseg.utils import get_palette

Hi, I used the solution you recommended, but I got an error when running "result = inference_model(model, gan_img_path)".
Error display "Error __init__() got an unexpected keyword argument 'img_scale'"

**Comment by LiangbinXie** [comment]
@nnmaitian This is a little bit weird. I'll check it this weekend.

**Comment by 201210302** [comment]
@nnmaitian I used the latest version of mmsegmentation, and modified artifact_detection.py as above to use the latest version of mmsegmentation. When editing, be sure to change not only the import part but also the function name used below. Accordingly, an error message may appear saying that the necessary files are not in the appropriate directory, but when I set the file path according to the error message, quick-inference was even executable.

**Comment by LiangbinXie** [comment]
@201210302 Hello, can you get the same results in this situation? If so, I will directly adopt your strategy.

**Comment by 201210302** [comment]
@nnmaitian  I uploaded the artifact_detection.py I used.

https://github.com/201210302/DeSRA_execution/tree/main/mmsegmentation/demo

**Comment by nnmaitian** [comment]
> @nnmaitian I uploaded the artifact_detection.py I used.[https://github.com/201210302/DeSRA_execution/tree/main/mmsegmentation/demo/artifact_detection.py](url)

Hi, thank you for your answer. Your link is invalid because the question was closed. I can't see your updated artifact_detection.py file. Can you post it directly under the question?

**Comment by nnmaitian** [comment]
> @nnmaitian This is a little bit weird. I'll check it this weekend.

Hello, is there any solution to the problem I encountered?

**Comment by LiangbinXie** [comment]
@nnmaitian I can check the code uploaded by @201210302. You should directly copy the link (not click) to a new webpage. Maybe you can try this modified file first.

**Comment by 201210302** [comment]
@nnmaitian I corrected the link

**Comment by haikunzhang95** [comment]
@201210302 I used your artifact_detection.py, but encountered:
**ModuleNotFoundError: No module named 'mmseg'**.
Do you have any suggestions?
Thank U.

**Comment by nnmaitian** [comment]
> @nnmaitian I corrected the link

@201210302 hi, thank you for your answer.Now my version is mmcv 2.0.0rc4, mmsegmentation v1.2.2. artifact_detection.py used your link. But I encountered the following problem.
![20240109-164129](https://github.com/TencentARC/DeSRA/assets/16910008/9873ca0c-066b-4909-9ef7-f14b703b896a)

**Comment by 201210302** [comment]
@nnmaitian @haikunzhang95 I will soon summarize the execution method I did and post it on my GitHub Readme. I will also re-upload the artifact_detection.py file used.

**Comment by 201210302** [comment]
@nnmaitian I updated the artifact_detection.py in my github https://github.com/201210302/DeSRA_execution/tree/main/mmsegmentation/demo

**Comment by 201210302** [comment]
@nnmaitian @LiangbinXie I upload the explanation of DeSRA execution: https://github.com/201210302/DeSRA_execution. Also, you need to redownload artifact_detection.py(I modified it)

**Comment by shahabe** [comment]
It seems in the newer version, the `inference_segmentor` has been changed to `inference_model`, based on this: https://github.com/open-mmlab/mmsegmentation/issues/3516#issuecomment-2071093363
